core/ir/aarch64/codec.c

```diff
@@ -3652,6 +3671,52 @@ encode_opnd_vindex_H(uint enc, int opcode, byte *pc, opnd_t opnd, OUT uint *enc_
     return true;
 }
 
+/* svemem_gpr_simm: 9 bit signed immediate offset added to base register defined in bits 5
+ * to 9.
+ */
+#if !defined(DR_HOST_NOT_TARGET) && !defined(STANDALONE_DECODER)
```

Comment:
Hi @derekbruening, there is an aspect of this SVE memory patch about which I would like clarification and advice.

The number of bytes in memory transfer instructions will depend on the size of the scalable vector on the SVE host. For the purposes of unit testing on non-SVE hardware including x86 we have fixed the vector length to 256 bits. 

This `#ifdef` clause sets `OPSZ_SVE_VL` for:
- use of SVE instructions in tools, clients, tracing etc, a value read from host h/w using `proc_get_vector_length()`, commented out for now (a patch will follow shortly).
- unit testing, a fixed value.

For the latter, how can we set the vector length for offline drdecode?
Some config or command line option or hard coded value?

Reply:
There is another similar issue with determining a runtime host parameter for drdecode -- let me find it to see if we had a solution.

Reply:
It was `dc zva`'s size which comes from the cache line size: #4400; PR #5265.

Generally for any of these "global" inputs to an instruction we'd have to add an API routine (similar to `dr_set_isa_mode`) that can be used to set this context for drdecode and we would probably also want a command line option for `drdisas`.  Maybe the only question is whether, if we are starting to accumulate multiple of them, we should combine them into an ISA config struct rather than having completely separate set and get routine?

Reply:
(The `dr_set_isa_mode` example is similar for ARM vs Thumb)

Reply:
In PR #5265 we proposed `set_dcache_zva_size()` for that one.  If we would prefer one config struct to make it easier to add more that makes sense to me; so we'd have something like `dr_set_isa_config(dr_isa_config_t *config)` or sthg with a size field so we can append.

Reply:
Direct link: https://github.com/DynamoRIO/dynamorio/pull/5265/files#r774769132

Probably we leave `dr_set_isa_mode` separate?  Or maybe there is also a mode field in the struct.

Reply:
@abhinav92003 for further opinions

Reply:
Should be fine for now with a TODO.